lib/symengine/basic.rb

```diff
@@ -11,8 +11,13 @@ def free_symbols
     end
     
     def abs
-        return SymEngine::abs(self)
+      SymEngine::abs(self)
+    end
+    def to_proc()
+      if self.free_symbols.count > 1
+        raise ArgumentError, "Too many free symbols! Only 1 allowed. Found #{self.free_symbols.count}."
```

Comment:
Actually here the problem was sending arguments to `to_proc` method. 
Because of the ordering problems discussed earlier, sending the symbol array is essential.

Is there a way to send arguments to the `to_proc` when it's called transparently(?) with the `&` operator(?).